### PR TITLE
Access Roles, Task 4: New Settings

### PR DESCRIPTION
**Pull request by benjaminjrobin**

### CODE REVIEW

**Comment by benjaminjrobin** [comment]
Meeting View Settings:
	HyperlinkNames
	ShowAddGuest
		User must be Attendance role
	AllowEditDescription
		User must be Attendance role
	ShowExtraValuesBox
		User must be Edit role
	ShowWandTargetBox
		User must be Attendance role
	ShowBlueToolbar
	ShowBlueToolbarIpadAttendance
	ShowBlueToolbarRollsheet
		MeetingDate must be greater than or equal to today
	DefaultEnableEdit
	ShowEnableBox
		User must be Attendance role
	ShowShowBox
	ShowAttendType
	ShowOtherAttend
	ShowCurrentMemberType
		User must be Attendance role

**Comment by benjaminjrobin** [comment]
Example XML for Meeting View Settings:
```
<?xml version="1.0" encoding="utf-8"?>
<roles>
    <role name="OrgLeadersOnly">
        <settings>
            <setting name="ShowSettingsTab" value="false" description="This setting determines whether or not the settings tab shows (like the existing UX-HideSettingsTabForOrgLeaders)" />
            <setting name="Meeting-HyperlinkNames" value="false" />
            <setting name="Meeting-ShowAddGuest" value="false" />
            <setting name="Meeting-AllowEditDescription" value="true" />
            <setting name="Meeting-ShowExtraValuesBox" value="true" />
            <setting name="Meeting-ShowWandTargetBox" value="true" />
            <setting name="Meeting-ShowBlueToolbar" value="true" />
            <setting name="Meeting-ShowBlueToolbarIpadAttendance" value="true" />
            <setting name="Meeting-ShowBlueToolbarRollsheet" value="true"" />
            <setting name="Meeting-DefaultEnableEdit" value="true" />
            <setting name="Meeting-ShowEnableBox" value="true" />
            <setting name="Meeting-ShowShowBox" value="true" />
            <setting name="Meeting-ShowAttendType" value="true" />
            <setting name="Meeting-ShowOtherAttend" value="true" />
            <setting name="Meeting-ShowCurrentMemberType" value="true" />
        </settings>
    </role>
</roles>

**Comment by benjaminjrobin** [comment]
Person View Setting:
```
<setting name="Person-ShowBlueToolbar" value="false" />

**Comment by benjaminjrobin** [comment]
Organization View Settings:
```
ShowOptionsBar
ShowSubgroupFiltersBar role= ManageGroup
ShowBirthday
ShowAddress
ShowCreateNew
	User must be in Attendance role
ShowDeleteMeeting
	User must be in Attendance role
ShowTagButtons

**Comment by benjaminjrobin** [comment]
Organization BlueToolbar Settings:
```
ShowBlueToolbar,
ShowBlueToolbarFullEmailMenu,
ShowBlueToolbarEmailMembers,
ShowBlueToolbarEmailProspects,
ShowBlueToolbarEmailMembersAndProspects,
ShowBlueToolbarExportMenu,
ShowBlueToolbarCustomReportsMenu,
ShowBlueToolbarAdminGearMenu,
ShowBlueToolbarSubGroupManagement,
ShowBlueToolbarMembersOnlyPage,
ShowBlueToolbarVolunteerCalendar,

File: CmsWeb/Areas/Org/Models/Org/OrgPeopleModel.cs [hunk 1]
```diff
@@ -258,15 +259,19 @@ public bool IsFiltered
         public bool HideInactiveButton => HttpContext.Current.User.IsInRole("OrgLeadersOnly") && DbUtil.Db.Setting("UX-HideInactiveOrgMembersForOrgLeadersOnly");
         public bool HidePendingButton => HttpContext.Current.User.IsInRole("OrgLeadersOnly") && DbUtil.Db.Setting("UX-HidePendingOrgMembersForOrgLeadersOnly");
         public bool HideGuestsButton => HttpContext.Current.User.IsInRole("OrgLeadersOnly") && DbUtil.Db.Setting("UX-HideGuestsOrgMembersForOrgLeadersOnly");
-        public bool HideOptionsButton => HttpContext.Current.User.IsInRole("OrgLeadersOnly") && DbUtil.Db.Setting("UX-OrgLeaderLimitedSearchPerson");
         public bool HideFilterOptions => HttpContext.Current.User.IsInRole("OrgLeadersOnly") && DbUtil.Db.Setting("UX-OrgLeaderLimitedSearchPerson");
 
+        public bool ShowOptions => RoleChecker.HasSetting(SettingName.Organization_ShowOptionsBar, true);
+        public bool ShowSubgroupFilters => RoleChecker.HasSetting(SettingName.Organization_ShowSubgroupFiltersBar, true);
```
Comment (drmohundro) [inline CmsWeb/Areas/Org/Models/Org/OrgPeopleModel.cs]: This should be replacing the existing `HideFilterOptions`.

File: CmsWeb/Views/Shared/Toolbar/Toolbar.cshtml [hunk 2]
```diff
@@ -1,40 +1,44 @@
...
+            if (Model is OrganizationModel && Model.ShowBlueToolbarAdminGear)
+            {
+                <div class="btn-group btn-group-lg dropdown-large">
+                    <button type="button" class="btn btn-primary dropdown-toggle" data-toggle="dropdown">
+                        <span class="fa fa-cog"></span>
+                    </button>
+                    @Html.Partial("Toolbar/Gear")
+                </div>
+
+            }
```
Comment (drmohundro) [inline CmsWeb/Views/Shared/Toolbar/Toolbar.cshtml]: I would suggest adding a variables section at the top like:

```cshtml
@{
  var showBlueToolbarAdminGear = true;
  // the others here, too
  if (Model is OrganizationModel) {
      showBlueToolbarAdminGear = Model.ShowBlueToolbarAdminGear;
      // the others here, too...
  }
}
```

File: CmsWeb/Web.config [hunk 3]
```diff
@@ -636,7 +636,7 @@
   <system.net>
     <mailSettings>
       <smtp deliveryMethod="SpecifiedPickupDirectory">
-        <specifiedPickupDirectory pickupDirectoryLocation="c:\email" />
+        <specifiedPickupDirectory pickupDirectoryLocation="C:\Users\Ben\Downloads" />
```
Comment (drmohundro) [inline CmsWeb/Web.config]: We'll want to remove these changes before we open the PR.

File: CmsWeb/Web.config [hunk 4]
```diff
@@ -39,7 +39,7 @@
     <!-- use (local) or .\SQLEXPRESS -->
     <add key="dbserver" value="(local)" />
     <!--Second half of db name like CMS_bellevue-->
-    <add key="host" value="bellevue" />
+    <add key="host" value="redeemer" />
```
Comment (drmohundro) [inline CmsWeb/Web.config]: We'll want to remove these changes before we open the PR.

File: CmsWeb/Views/Shared/Toolbar/Toolbar.cshtml (hunk 2 above)
Comment (benjaminjrobin) [inline CmsWeb/Views/Shared/Toolbar/Toolbar.cshtml]: Done.

**Comment by drmohundro** [comment]
Closing this one in lieu of #81

**Pull request closed without merging**